### PR TITLE
firedancer: fix epoch transition and pack topo

### PR DESCRIPTION
**Pull request by lheeger-jump**
Fixes an issue where we push stakes too early to poh tile and where pack gets corrupted frags and fixes some status cache issues.

### CODE REVIEW

**Review by arjain4 — APPROVED** [review]

File: src/disco/poh/fd_poh_tile.c
Comment (ptaffet-jump) [inline src/disco/poh/fd_poh_tile.c:45]: Get rid of these warnings before merging, right?

Comment (arjain4) [reply src/disco/poh/fd_poh_tile.c:45]: We are actively debugging using this branch right now, making it a draft

File: src/disco/bank/fd_txncache.c
Comment (arjain4) [inline src/disco/bank/fd_txncache.c:3]: let's get rid of this

File: src/disco/bank/fd_txncache.c
Comment (lidatong) [inline src/disco/bank/fd_txncache.c:353]: need FD_TXNCACHE_XXX constants for the `(ULONG_MAX - 1UL)`, `-2UL` business that describe what's going on and documenting that 

i won't block this PR on that but please add

File: src/disco/bank/fd_txncache.c
Comment (lheeger-jump) [reply src/disco/bank/fd_txncache.c]: done

**Review by lidatong — APPROVED** [review]

**Review by arjain4 — APPROVED** [review]

File: src/disco/bank/fd_txncache.c
Comment (lheeger-jump) [reply src/disco/bank/fd_txncache.c:353]: @arjain4 can you do this in your slotcache PR?

Comment (arjain4) [reply src/disco/bank/fd_txncache.c:353]: yep have those changes in mind

**Pull request merged**
